packages/compiler-cli/ngcc/test/writing/in_place_file_writer_spec.ts

```diff
@@ -29,77 +29,140 @@ runInEachFileSystem(() => {
-         const fs = getFileSystem();
-         const logger = new MockLogger();
-         const fileWriter = new InPlaceFileWriter(fs, logger, /* errorOnFailedEntryPoint */ true);
-         const absoluteBackupPath = _('/package/path/already-backed-up.js');
-         expect(
-             () => fileWriter.writeBundle(
-                 {} as EntryPointBundle,
-                 [{path: absoluteBackupPath, contents: 'MODIFIED BACKED UP'}]))
-             .toThrowError(`Tried to overwrite ${
-                 absoluteBackupPath}.__ivy_ngcc_bak with an ngcc back up file, which is disallowed.`);
-       });
-
-    it('should log an error, and skip writing the file, if the backup file already exists and errorOnFailedEntryPoint is false',
-       () => {
-         const fs = getFileSystem();
-         const logger = new MockLogger();
-         const fileWriter = new InPlaceFileWriter(fs, logger, /* errorOnFailedEntryPoint */ false);
-         const absoluteBackupPath = _('/package/path/already-backed-up.js');
-         fileWriter.writeBundle(
-             {} as EntryPointBundle, [{path: absoluteBackupPath, contents: 'MODIFIED BACKED UP'}]);
-         // Should not have written the new file nor overwritten the backup file.
-         expect(fs.readFile(absoluteBackupPath)).toEqual('ORIGINAL ALREADY BACKED UP');
-         expect(fs.readFile(_(absoluteBackupPath + '.__ivy_ngcc_bak'))).toEqual('BACKED UP');
-         expect(logger.logs.error).toEqual([[
-           `Tried to write ${
-               absoluteBackupPath}.__ivy_ngcc_bak with an ngcc back up file but it already exists so not writing, nor backing up, ${
-               absoluteBackupPath}.\n` +
-           `This error may be because two or more entry-points overlap and ngcc has been asked to process some files more than once.\n` +
-           `You should check other entry-points in this package and set up a config to ignore any that you are not using.`
-         ]]);
-       });
+    describe('revertFile()', () => {
+      it('should revert a written file (and its backup)', () => {
+        const fs = getFileSystem();
+        const logger = new MockLogger();
+        const fileWriter = new InPlaceFileWriter(fs, logger, /* errorOnFailedEntryPoint */ true);
+
+        const packagePath = _('/package/path');
+        const filePath = _('/package/path/top-level.js');
+        const fileBackupPath = _(`/package/path/top-level.js${NGCC_BACKUP_EXTENSION}`);
+
+        fileWriter.writeBundle({} as EntryPointBundle, [
+          {path: filePath, contents: 'MODIFIED TOP LEVEL'},
+        ]);
+        expect(fs.readFile(filePath)).toBe('MODIFIED TOP LEVEL');
+        expect(fs.readFile(fileBackupPath)).toBe('ORIGINAL TOP LEVEL');
+
+        fileWriter.revertFile(filePath, packagePath);
+        expect(fs.readFile(filePath)).toBe('ORIGINAL TOP LEVEL');
+        expect(fs.exists(fileBackupPath)).toBeFalse();
+      });
+
+      it('should only revert the written file if there is no backup', () => {
```

Comment:
This test description is a bit misleading, since we do revert the file (to nothing). The implication of not reverting would be to leave the file untouched, which is not what happens. Perhaps instead it should be something like:

```suggestion
      it('should just remove the written file if there is no backup', () => {
```